### ISSUE TITLE
Add optional case sensitivity to Should[Not]StartWith and Should[Not]EndWith

### ISSUE THREAD

**Issue by chaitanyagurrapu** [shouldly/shouldly#206]
Currently the string.ShouldBe extension method takes an optional case sensitivity parameter to determine whether to do a case sensitive comparison or a case insensitive comparison.

This option should also be extended to "ShouldStartWith", "ShouldNotStartWith", "ShouldEndWith" and "ShouldNotEndWith" methods. Currently these methods only perform a case insensitive match.

**Comment by JakeGinnivan** [comment]
:+1: we need to make sure that case sensitive is opt in so this is not a breaking change

**Comment by JoeMighty** [comment]
Presumably this is going to dramatically increase the number of methods for the aforementioned Shouldly methods? At the moment (if we're going on the changes in the AdditionalInfoMessages branch) there are 3 methods per Shouldly test (2 of which are overloads), this would take it to 6 per test method?

**Comment by JakeGinnivan** [comment]
It will, it would be nice to use optional parameters for this. Technically that is a breaking change though.

Maybe we can try and consolidate the overloads which can be simplified with optional parameters as well, then bump to v3?

**Comment by JoeMighty** [comment]
Yeah, optional parameters are an option. Though wouldn't making them optional introduce an inconsistency in the api? ShouldBe's caseSensitivity argument aren't optional. Perhaps it's worth making ShouldBe optional too?

**Comment by JakeGinnivan** [comment]
The reason they were not made optional is that is a binary breaking change in the API. We probably don't need binary compatibility with a assertion framework though.. Might ask twitter

**Comment by JakeGinnivan** [comment]
Consensus seems to be binary. Lets not use optional parameters and then we can release 2.5 instead of 3.0. We are going for more overloads with the custom message, so lets do more overloads in this case too.

@chaitanyagurrapu could you change it to use overloads instead? Once it is merged we will need to add another task on #195 to add optional parameters for the overloads with custom messages (they are new overloads so no breaking change adding optional params to them)

**Comment by JoeMighty** [comment]
@JakeGinnivan I see, that makes sense.

**Comment by chaitanyagurrapu** [comment]
Done. Hope that is what you were looking for.

**Comment by JakeGinnivan** [comment]
After making you go to that effort. I realised my moving methods around in the custom messages PR is a breaking change. Am going to bump the next release to v3

Sorry about that

**Comment by JakeGinnivan** [comment]
I have integrated this change with the custom message PR.

We should probably do this for ShouldContain and other string related methods
